cpp_exercise/src/lib/ipcmsgqlib.cc

```diff
@@ -61,9 +73,27 @@ MsgQueueSender::MsgQueueSender(const std::string & filename): _file_name(filenam
             //reading and sending the data of file in smaller parts of 4096
             std::vector<unsigned char> buffer;
             buffer = fd.readFile();
+            
+            clock_gettime(CLOCK_REALTIME, &_ts);
+
+            //setting the time for 10 seconds so if no data received after 10 seconds it will break the loop   
+     	    _ts.tv_sec += 10;	
   
 
-            mq_send(_msg_queue, reinterpret_cast<char *>(buffer.data()),buffer.size(),_priority);
+           
+            ret = mq_timedsend(_msg_queue,(char*)buffer.data(),buffer.size(),_priority, &_ts);
+
+            if(ret<0){
+                if (errno == ETIMEDOUT){
+                std::cout<<"          "<<strerror(errno)<<std::endl;
+                throw IPCException("          PCSender ERROR: Cannot Connect to the Receiver");
```

Comment:
You can put `strerror(errno)` inside throw(), errno will be printed together with the error message that way

Reply:
Fixed